### PR TITLE
[YAML] Java PubSubIO

### PR DESCRIPTION
**Pull request by robertwb**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @chamikaramj

**Comment by robertwb** [comment]
Ping on this PR.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubReadSchemaTransformProvider.java [hunk 1]
```diff
@@ -89,38 +87,43 @@ public SchemaTransform from(PubsubReadSchemaTransformConfiguration configuration
...
 
-    Schema beamSchema;
-    SerializableFunction<byte[], Row> valueMapper;
+    Schema payloadSchema;
+    SerializableFunction<byte[], Row> payloadMapper;
 
-    if (!VALID_DATA_FORMATS.contains(configuration.getFormat())) {
+    String format =
+        configuration.getFormat() == null ? null : configuration.getFormat().toUpperCase();
+    if (Objects.equals(format, "RAW")) {
```
Comment (chamikaramj) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubReadSchemaTransformProvider.java:107]: Just do `if ("RAW".equals(format))` ? (and similarly below)

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubReadSchemaTransformProvider.java [hunk 2]
```diff
@@ -207,26 +271,47 @@ PubsubIO.Read<PubsubMessage> buildPubsubRead() {
...
+      if (!Strings.isNullOrEmpty(configuration.getTimestampAttribute())) {
+        pubsubRead = pubsubRead.withTimestampAttribute(configuration.getTimestampAttribute());
+      }
       return pubsubRead;
     }
 
     @Override
     public PCollectionRowTuple expand(PCollectionRowTuple input) {
       PubsubIO.Read<PubsubMessage> pubsubRead = buildPubsubRead();
+      @SuppressWarnings("nullness")
```
Comment (chamikaramj) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubReadSchemaTransformProvider.java:286]: Is this suppression needed ?(if so probably can be removed by adding a null check before this line).

**Review by chamikaramj — COMMENTED** [review]
Thanks. LGTM.

Just nits.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubReadSchemaTransformProvider.java (hunk 2 above)
Comment (robertwb) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubReadSchemaTransformProvider.java:286]: Unfortunately the nullness checker is not smart enough for that.

**Review by Polber — APPROVED** [review]

**Pull request merged**